source/geod24/concurrency.d

```diff
@@ -985,6 +986,161 @@ protected:
         bool notified;
     }
 
+    class Channel (T)
+    {
+        this(ulong size = 0)
```

Comment:
I'm guessing if size = 0 it changes the behavior of the channel. Buffered vs unbuffered. This needs to be documented.

And in fact I think having a default value is probably not a good idea.

Reply:
Yes you are correct. I gave it a default value to be close to go channel syntax. Should we require caller to explicitly give 0 or have an overload for it?

Reply:
> go channel syntax

Alright, I wasn't aware. A sensible default is also ok, as long as it's documented as such.

Reply:
Default is unbuffered and I think that's sane. It means you always context switch when writing, and context switch on reading if there's no message. If the channel is buffered, you might, or might not, context switch on write.